pennylane/utils.py

```diff
@@ -323,6 +326,22 @@ def circuit2():
             + ",".join(string_reps)
         )
 
+    if qml.tape_mode_active():
+        for op in operation_list:
+            try:
+                qml.tape.QueuingContext.remove(op)
+            except KeyError:
+                pass
+
+        with qml.tape.QuantumTape() as tape:
```

Comment:
How come we are removing the operations before this step if we'd actually construct a new tape?

Reply:
We need to create a temporary tape for the operations we want to invert, since the logic for inversion is a tape method.

This is just a temporary fix, and we should replace this with a proper functional version